coderd/util/apiversion/apiversion_test.go

```diff
@@ -0,0 +1,85 @@
+package apiversion_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/v2/coderd/util/apiversion"
+)
+
+func TestAPIVersionValidate(t *testing.T) {
+	// Given
+	v := apiversion.New([]int{2, 1}, 0)
```

Comment:
since this is now abstract, set the minor to 1 so you can also test that it works for minor < supportedMinor